### DISCUSSION TITLE
Roadmap for Snowflake Open Catalog Terraform

### DISCUSSION THREAD

**Discussion by juzov-billie** [Snowflake-Labs/terraform-provider-snowflake#3156] [Q&A]
Hi as the product now is in GA, can we expect to terraform the service at some point as well?

**Comment by sfc-gh-asawicki** [comment]
Hey @juzov-billie. Thanks for reaching out to us.

Currently, our provider supports only managing objects that can be managed by SQ. AFAIK, currently, there are no SQL statements to set up open catalogs.

**Reply by juzov-billie** [reply]
Got it, and another provider is not on the roadmap, as far as you know?

**Reply by sfc-gh-asawicki** [reply]
I have just browsed one more time through the docs. Is this: https://other-docs.snowflake.com/en/opencatalog/create-open-catalog-account#create-an-account-by-using-snowflake-sql what you are asking for?

**Reply by juzov-billie** [reply]
That and, anything you can create in the account you create. Setting up ip rules, creating roles and users, catalogs etc. Think they would be fairly suited to be terraform resources.

**Reply by sfc-gh-asawicki** [reply]
As I understand the instructions from here: https://other-docs.snowflake.com/en/opencatalog/create-catalog, to set up the catalog, you have to first manage multiple objects on your Amazon/Google/Azure accounts. These objects can be managed by the dedicated providers (like https://registry.terraform.io/providers/hashicorp/aws/latest/docs).

The part "Use Open Catalog to create a catalog." (e.g. https://other-docs.snowflake.com/en/opencatalog/create-catalog#step-3-create-a-catalog) as I understand is currently only available through GUI and not SQL.

**Reply by juzov-billie** [reply]
yeah think it would need to be rest api based

**Reply by sfc-gh-asawicki** [reply]
Yup, so it boils down to:
- adding not only SQL-based communication to our provider (mentioned above)
- making sure that there is a REST API that can be easily consumed (APIs for GUIs are hardly ever a good fit for such programmatic consumption)

**Reply by juzov-billie** [reply]
The underlying catalog api, is rest afaik. So its partially there. https://polaris.io/index.html#tag/Catalog-API/operation/updateProperties

**Reply by juzov-billie** [reply]
The snowflake provider, should on the snowflake side at least allow catalog integrations https://docs.snowflake.com/en/sql-reference/sql/create-catalog-integration-open-catalog

**Reply by juzov-billie** [reply]
I looked into the api requests that are being done for e.g. network policies, and this is simply a rest request with a sql body.

**Reply by sfc-gh-asawicki** [reply]
https://docs.snowflake.com/en/sql-reference/sql/create-catalog-integration-open-catalog will be added as a resource to the provider but not before the V1.

**Reply by ghost** [reply]
Since V1 has been released, do we have any further details on the planning now?

**Reply by sfc-gh-asawicki** [reply]
Hey @oteller2. We won't tackle the feature gap before the GA (check https://github.com/Snowflake-Labs/terraform-provider-snowflake/blob/main/ROADMAP.md#feature-gap-closing---the-current-approach). We can't commit to any timelines after the GA yet.
